### PR TITLE
ci: fix release workflow permissions

### LINKED ISSUE #30: 🛠 Tooling: Release job blocked from pushing to GitHub
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/ts-api-utils/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

This GitHub Action `release.yml` Release job failed to push to GitHub: https://github.com/JoshuaKGoldberg/ts-api-utils/actions/runs/4115398512/jobs/7104053426

```plaintext
 M package.json
ERROR remote: Permission to JoshuaKGoldberg/ts-api-utils.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/JoshuaKGoldberg/ts-api-utils/': The requested URL returned error: 403
```

A later run that fails to publish a package version to npm twice (since the Git commit that bumped version never was pushed to GitHub): https://github.com/JoshuaKGoldberg/ts-api-utils/actions/runs/4115932211/jobs/7105309725

```plaintext
npm notice Publishing to https://registry.npmjs.org/
npm ERR! code E403
npm ERR! 403 403 Forbidden - PUT https://registry.npmjs.org/ts-api-tools - You cannot publish over the previously published versions: 0.0.18.
npm ERR! 403 In most cases, you or one of your dependencies are requesting
npm ERR! 403 a package version that is forbidden by your security policy, or
npm ERR! 403 on a server you do not have access to.
```

...but I set up a dev PAT token. Why is this broken?

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by RebeccaStevens**
fix #30

## PR Checklist

- [x] Addresses an existing open issue: fixes #30
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/ts-api-utils/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/ts-api-utils/blob/main/.github/CONTRIBUTING.md) were taken

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -5,6 +5,9 @@ on:
     branches:
       - main
 
+permissions:
+  contents: write
```
Comment (RebeccaStevens) [inline .github/workflows/release.yml:9]: Give the default token permission to make changes

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -22,7 +25,6 @@ jobs:
       - name: Delete branch protection on main
         uses: actions/github-script@v6.4.0
         with:
-          github-token: ${{ secrets.ACCESS_TOKEN }}
```
Comment (RebeccaStevens) [inline .github/workflows/release.yml:25]: Use the default token instead of a PAT

**Review by JoshuaKGoldberg — APPROVED** [review]
Thank you!! I really do not like investigating this kind of thing 😂

**Comment by JoshuaKGoldberg** [comment]
What a lovely, ridiculously overloaded release! https://github.com/JoshuaKGoldberg/ts-api-utils/releases/tag/0.0.24

**Pull request merged**